src/test/java/com/epam/test_generator/services/ProjectServiceTest.java

```diff
@@ -0,0 +1,398 @@
+            userDTOSet
+        );
+        simpleProject1DTO.setId(2L);
+        simpleProject2DTO = new ProjectDTO(
+            "project2",
+            "project2",
+            true,
+            userDTOSet
+        );
+        simpleProject2DTO.setId(2L);
+        simpleProjectFullDTO1 = new ProjectFullDTO(
+            "project1",
+            "project1",
+            null,
+            userDTOSet,
+            true
+        );
+        simpleProjectFullDTO1.setId(1L);
+
+        userSet = Stream.of(simpleUser1, simpleUser2)
+            .collect(Collectors.toSet());
+        simpleProject1.setUsers(userSet);
+        simpleProject2.setUsers(userSet);
+        userDTOSet = Stream.of(simpleUserDTO1, simpleUserDTO2)
+            .collect(Collectors.toSet());
+        simpleProject1DTO.setUsers(userDTOSet);
+        simpleProject2DTO.setUsers(userDTOSet);
+        simpleProjectFullDTO1.setUsers(userDTOSet);
+        expectedProjectDTOs = Stream.of(simpleProject1DTO, simpleProject2DTO)
+            .collect(Collectors.toList());
+        expectedProjects = Stream.of(simpleProject1, simpleProject2)
+            .collect(Collectors.toList());
+    }
+
+    @Test
+    public void getProjects_expectedProjectsList() {
+        when(projectDAO.findAll()).thenReturn(expectedProjects);
+        final List<ProjectDTO> projects = projectService.getProjects();
+        assertEquals(projects, projects);
+
+        verify(projectDAO).findAll();
+
+    }
+
+    @Test
+    public void getAuthUserProjects_expectedUserProjectsList() {
+        when(authentication.getPrincipal()).thenReturn(new AuthenticatedUser(
+            null,
+            simpleUser1.getEmail(),
+            null,
+            null,
+            false
+        ));
+        when(userService.getUserByEmail(simpleUser1.getEmail())).thenReturn(simpleUser1);
+        when(userService.getUserById(simpleUser1.getId())).thenReturn(simpleUser1);
+        when(projectService.getProjectsByUserId(simpleUser1.getId())).thenReturn(expectedProjects);
+        when(projectTransformer.toDtoList(expectedProjects)).thenReturn(expectedProjectDTOs);
+        final List<ProjectDTO> actualProjectDTOs = projectService
+            .getAuthUserProjects(authentication);
+        assertEquals(expectedProjectDTOs, actualProjectDTOs);
+
+        verify(userService).getUserByEmail(eq(simpleUser1.getEmail()));
+        verify(projectTransformer).toDtoList(eq(expectedProjects));
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void getAuthUserProjects_expectedUserNotFoundException() {
+        when(authentication.getPrincipal()).thenReturn(new AuthenticatedUser(
+            null,
+            simpleUser1.getEmail(),
+            null,
+            null,
+            false
+        ));
+        when(userService.getUserByEmail(simpleUser1.getEmail())).thenReturn(null);
+        final List<ProjectDTO> actualProjectDTOs = projectService
+            .getAuthUserProjects(authentication);
```

Comment:
final List<ProjectDTO> actualProjectDTOs
this declaration can be removed.

Reply:
removed (and in other places too)